### PR TITLE
docs: add more detail about reviewing image promotion PR

### PR DESCRIPTION
**Pull request by jackfrancis**
**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:

This PR adds some supporting material to help folks better review the image promotion PR during the release process.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: docs/book/src/developers/releasing.md [hunk 1]
```diff
@@ -95,7 +95,13 @@ This will automatically trigger a [Github Action](https://github.com/kubernetes-
   - `export GITHUB_TOKEN=<your GH token>`
   - `make promote-images`
 
-This will automatically create a PR in [k8s.io](https://github.com/kubernetes/k8s.io) and assign the CAPZ maintainers. Example PR: https://github.com/kubernetes/k8s.io/pull/3007.
+This will automatically create a PR in [k8s.io](https://github.com/kubernetes/k8s.io) and assign the CAPZ maintainers. Example PR: https://github.com/kubernetes/k8s.io/pull/4284.
+
+For reviewers of the above-created PR, to confirm that the resultant image SHA-to-tag addition is valid, you can check against the Google Container Registry.
```
Comment (CecileRobertMichon) [inline docs/book/src/developers/releasing.md:100]:
```suggestion
For reviewers of the above-created PR, to confirm that the resultant image SHA-to-tag addition is valid, you can check against the [staging repository](https://console.cloud.google.com/gcr/images/k8s-staging-cluster-api-azure/GLOBAL/cluster-api-azure-controller?rImageListsize=30)
```

To be consistent with the wording above

Comment (CecileRobertMichon) [inline docs/book/src/developers/releasing.md:100]: should we make it clear that this isn't a required step for review given that the SHA is auto-generated by `kpromo`? If the tool ever produces a PR that has a SHA mismatching the tag that would be a bug, not human error.

Comment (jackfrancis) [reply docs/book/src/developers/releasing.md:100]: What if someone ran `make promote-images`, then later on before the image PR is merged, found out that the release was tagged to the wrong commit, deleted the tag, then re-created against a new commit?

Comment (CecileRobertMichon) [reply docs/book/src/developers/releasing.md:100]: fair :)

Comment (jackfrancis) [reply docs/book/src/developers/releasing.md]: done

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/approve

thanks!

**Pull request merged**